File: tests/fixtures/graph-src-tier3/swift/Shapes.swift
import Foundation

typealias Scalar = Double

let pi: Scalar = 3.14159

/// Free function.
func circleArea(radius: Scalar) -> Scalar {
    return radius * radius * pi
}

protocol Shape {
    func area() -> Scalar
}

struct Point {
    var x: Scalar
    var y: Scalar

    func distanceTo(other: Point) -> Scalar {
        return hypot(x - other.x, y - other.y)
    }
}

class Base {
    var tag: Scalar = 0
}

class Circle: Base, Shape {
    var radius: Scalar

    init(radius: Scalar) {
        self.radius = radius
    }

    func area() -> Scalar {
        return circleArea(radius: radius)
    }
}

enum Suit {
    case hearts
    case spades
}

func hypot(_ a: Scalar, _ b: Scalar) -> Scalar {
    return (a * a + b * b).squareRoot()
}

File: tests/fixtures/graph-src-tier3/kotlin/Shapes.kt
package com.example.geo

import kotlin.math.sqrt

const val PI: Double = 3.14159

typealias Scalar = Double

/** Free function. */
fun circleArea(radius: Scalar): Scalar {
    return radius * radius * PI
}

interface Shape {
    fun area(): Scalar
}

data class Point(val x: Scalar, val y: Scalar) {
    fun distanceTo(other: Point): Scalar {
        return hypot(x - other.x, y - other.y)
    }
}

class Circle(val radius: Scalar) : Shape {
    override fun area(): Scalar {
        return circleArea(radius)
    }
}

enum class Suit {
    HEARTS,
    SPADES
}

object Registry {
    fun count(): Int = 0
}

fun Scalar.doubled(): Scalar = this * 2

fun hypot(a: Scalar, b: Scalar): Scalar = sqrt(a * a + b * b)
